### PR TITLE
Added native SIGN? func.

### PR DESCRIPTION
**Pull request by greggirwin**

### CODE REVIEW

File: runtime/natives.reds [hunk 1]
```diff
@@ -1474,6 +1474,40 @@ natives: context [
...
+	sign?*: func [
+		check?  [logic!]
+		return: [red-integer!]
+		/local
+			i   [red-integer!]
+			f	[red-float!]
+			res [red-value!]
+			ret [integer!]
+	][
+		#typecheck -sign?-								;-- `sign?` would be replaced by lexer
```
Comment (dockimbel) [inline runtime/natives.reds]: Decorating `sign?` there is not necessary, as it's only used when the word could conflict with an existing R/S definition, in such case, a decorated/undecorated pair needs to be added to this function: https://github.com/red/red/blob/master/compiler.r#L3652

You can remove the decoration in another PR, I will accept this one as-is.

Comment (greggirwin) [inline runtime/natives.reds]: Thanks! Learning all the time.

Comment (iArnold) [inline runtime/natives.reds]: @greggirwin Another Guru meditation coming up? And how does one quickly check if a conflict with an existing R/S definition occurs?

Comment (greggirwin) [inline runtime/natives.reds]: I added the primary note, but don't have a solution for quickly checking names yet.

**Pull request closed without merging**